examples/src/test/java/jdocs/eventsourced/ShoppingCart.java

```diff
@@ -299,27 +295,37 @@ public String toString() {
 
   public static void init(ActorSystem<?> system, int numberOfEventProcessors) {
     ClusterSharding.get(system).init(Entity.of(ENTITY_TYPE_KEY, entityContext -> {
-      int n = Math.abs(entityContext.getEntityId().hashCode() % numberOfEventProcessors);
-      String eventProcessorTag = "carts-" + n;
-      return ShoppingCart.create(entityContext.getEntityId(), Collections.singleton(eventProcessorTag));
+      return ShoppingCart.create(entityContext.getEntityId());
     })
       .withRole("write-model"));
   }
 
-  public static Behavior<Command> create(String cartId, Set<String> eventProcessorTags) {
-    return new ShoppingCart(cartId, eventProcessorTags);
+  public static Behavior<Command> create(String cartId) {
+    return new ShoppingCart(cartId);
   }
 
   private final String cartId;
-  private final Set<String> eventProcessorTags;
 
-  private ShoppingCart(String cartId, Set<String> eventProcessorTags) {
+  private ShoppingCart(String cartId) {
     super(PersistenceId.of(ENTITY_TYPE_KEY.name(), cartId),
       SupervisorStrategy.restartWithBackoff(Duration.ofMillis(200), Duration.ofSeconds(5), 0.1));
     this.cartId = cartId;
-    this.eventProcessorTags = eventProcessorTags;
   }
 
+  //#slicingTags
+  public final static List<String> tags =
+    Collections.unmodifiableList(Arrays.asList("carts-0", "carts-1", "carts-2", "carts-3", "carts-4"));
+  //#slicingTags
+
+  //#tagging
+  @Override
+  public Set<String> tagsFor(Event event) {
+    int n = Math.abs(event.getCartId().hashCode() % tags.size());
+    String selectedTag = tags.get(n);
+    return Collections.singleton(selectedTag);
+  }
```

Comment:
Since the `cartId` is always the same for the entity it might be better to define this as a final instance field, assigned in the constructor. I mean, also good to highlight that it's always the same for the same entityId.

Reply:
It is already, but somehow I ignore it and picked the one in the event. 
I don't know why we have it there, maybe we use it in the akka-samples on the read-side. 

I will leave it in the events and use the final field in the model.

Reply:
ok, that works, but the reader might still not see that it's per entity and not per event.

```
private final Set<String> tags;
{
  int n = Math.abs(cartId.hashCode() % tags.size());
  String selectedTag = tags.get(n);
  tags = Collections.singleton(selectedTag);
}

public Set<String> tagsFor(Event event) {
  return tags;
}
```

Reply:
ok, but using a `{}` block is tricky. Compiler complains that cartId may not have been initialised. We need to pick the selectedTag in the constructor. That will increate the code fragment.

Reply:
Isn't that clear enough if we reference the cart as `this.cartId`?

Reply:
ok, leave it as is